### PR TITLE
Conflicting Area Tag Combination Check

### PR DESCRIPTION
**Pull request by danielduhh**
Flags Area objects with mutually exclusive tags

### CODE REVIEW

File: docs/checks/conflictingAreaTagCombination.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    return object instanceof Area
+        && Validators.isNotOfType(object, AreaTag.class, AreaTag.NO);
+}
+```
+
+Using the Validators class, we store each conflicting combination into a [Predicate](https://docs.oracle.com/javase/8/docs/api/java/util/function/Predicate.html) variable that can be used to test its truthiness.
+```java
+private static final Predicate<Taggable> NATURAL_WATER_MANMANDE = object -> 
+        Validators.isOfType(object, NaturalTag.class, NaturalTag.WATER)
+        && Validators.isNotOfType(object, ManMadeTag.class, ManMadeTag.RESERVOIR_COVERED, ManMadeTag.WASTEWATER_PLANT);;
```
Comment (Bentleysb) [inline docs/checks/conflictingAreaTagCombination.md]: Extra ;

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingAreaTagCombination.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    // Natural tag should not exist with any highway tags.
+    private static final Predicate<Taggable> NATURAL_HIGHWAY = object -> Validators.hasValuesFor(
+            object, NaturalTag.class) && Validators.hasValuesFor(object, HighwayTag.class);
+    // The natural=WATER tag should not exist with any man_made tags. The exceptions are
+    // man_made=RESERVOIR_COVERED and man_made=WASTEWATER_PLANT.
+    private static final Predicate<Taggable> NATURAL_WATER_MANMANDE = object -> Validators
+            .isOfType(object, NaturalTag.class, NaturalTag.WATER)
+            && Validators.isNotOfType(object, ManMadeTag.class, ManMadeTag.RESERVOIR_COVERED,
+                    ManMadeTag.WASTEWATER_PLANT);
+    // The natural=WATER tag should exist with any landuse tags. The exceptions are landuse=BASIN,
```
Comment (Bentleysb) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingAreaTagCombination.java]: should exist -> should not exist

File: src/test/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingAreaTagCombinationCheckTestRule.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    private Atlas waterLandUseTagAtlas;
+
+    // natural=water and landuse=aquaculture
+    @TestAtlas(areas = { @Area(id = INVALID_AREA_ID, coordinates = {
+            @Loc(value = AREA_LOCATION_ONE), @Loc(value = AREA_LOCATION_TWO),
+            @Loc(value = AREA_LOCATION_THREE),
+            @Loc(value = AREA_LOCATION_ONE) }, tags = { "natural=water", "landuse=aquaculture" }) })
+    private Atlas waterLandUseAquacultureAtlas;
+
+    // natural=water and man_made=
```
Comment (Bentleysb) [inline src/test/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingAreaTagCombinationCheckTestRule.java]: man_made=?

**Review by Bentleysb — CHANGES_REQUESTED** [review]
Just a few doc/comment typos.

File: docs/checks/conflictingAreaTagCombination.md (hunk 1 above)
Comment (danielduhh) [reply docs/checks/conflictingAreaTagCombination.md]: yes

File: src/test/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingAreaTagCombinationCheckTestRule.java (hunk 3 above)
Comment (danielduhh) [reply src/test/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingAreaTagCombinationCheckTestRule.java]: yup

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingAreaTagCombination.java (hunk 2 above)
Comment (danielduhh) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingAreaTagCombination.java]: yup

File: src/test/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingAreaTagCombinationCheckTestRule.java (hunk 3 above)
Comment (danielduhh) [reply src/test/java/org/openstreetmap/atlas/checks/validation/tag/ConflictingAreaTagCombinationCheckTestRule.java]: word

**Review by Bentleysb — APPROVED** [review]
Looks good to me. 

The only improvement that comes to mind is making the conflicting tags configurable. However, this would mean a near total rewrite, I believe.

**Pull request closed without merging**